src/tests/tribol_enzyme_poly_intersect.cpp

```diff
@@ -0,0 +1,308 @@
+// Copyright (c) 2017-2023, Lawrence Livermore National Security, LLC and
+// other Tribol Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (MIT)
+
+//-----------------------------------------------------------------------------
+//
+// file: tribol_enzyme_poly_intersect.cpp
+//
+//-----------------------------------------------------------------------------
+
+#include <iostream>
+
+#include "tribol/config.hpp"
+
+#include "gtest/gtest.h"
+
+#ifdef TRIBOL_USE_UMPIRE
+#include "umpire/ResourceManager.hpp"
+#endif
+
+#include "mfem.hpp"
+
+#include "tribol/common/Enzyme.hpp"
+#include "tribol/geom/GeomUtilities.hpp"
+
+namespace tribol {
+
+/**
+ * @brief Test fixture for the Enzyme-based derivatives of intersection polynomial calculations.
+ */
+class EnzymePolyIntersectTest : public testing::Test {
+ protected:
+  static constexpr double delta_{ 1.0e-8 };
+  void SetUp() override {}
+
+  void CheckIntersection( RealT* x1, RealT* x2, int* stencil_dir, RealT fd_tol = delta_, RealT pos_tol = 1.0e-8,
+                          RealT len_tol = 1.0e-8, [[maybe_unused]] std::string name = "" )
+  {
+    RealT xi[16];
+    OverlapVertexType type[8];
+    IndexT edge1[8];
+    IndexT edge2[8];
+    for ( int i{ 0 }; i < 8; ++i ) {
+      xi[i] = 0.0;
+      xi[i + 8] = 0.0;
+      type[i] = OverlapVertexType::A;
+      edge1[i] = -1;
+      edge2[i] = -1;
+    }
+    auto num_poly_verts = 0;
+    auto d_num_poly_verts = 0;
+    auto d_pos_tol = 0.0;
+    auto d_len_tol = 0.0;
+    RealT area = 0.0;
+    Intersection2DPolygon( x1, x1 + 4, 4, x2, x2 + 4, 4, pos_tol, len_tol, xi, xi + 8, num_poly_verts, area, true, type,
+                           edge1, edge2 );
+    std::cout << std::setprecision( 15 ) << "Element 1 coords" << std::endl;
+    for ( int i{ 0 }; i < 4; ++i ) {
+      std::cout << "(" << x1[i] << ", " << x1[i + 4] << ")\n";
+    }
+    std::cout << std::setprecision( 15 ) << "Element 2 coords" << std::endl;
+    for ( int i{ 0 }; i < 4; ++i ) {
+      std::cout << "(" << x2[i] << ", " << x2[i + 4] << ")\n";
+    }
+    std::cout << std::setprecision( 15 ) << "Number of vertices: " << num_poly_verts << "   Polygon area: " << area
+              << std::endl;
+
+    for ( int i{ 0 }; i < num_poly_verts; ++i ) {
```

Comment:
This output looks a little cryptic to me. Can you clarify?

Reply:
Added comments elsewhere, so hopefully this is clarified.